express.js

```diff
@@ -376,7 +400,356 @@ module.exports = (app) => {
+				} else if(!functionMatch || functionMatch > 1) {
+					status.scoreModule = 'fail'
+					action.scoreModule = "'check_answer' function was not defined or defined more than once"
+				} else {
+					status.scoreModule = 'pass'
+				}
+
+			} catch(error) {
+				status.scoreModule = 'fail'
+				action.scoreModule = "score module missing or can't be parsed"
+			}
+		} else {
+			//if we can't get the name of the score module we need, we can't check validity
+			//this shouldn't ever happen if the whole install.yaml check block passes
+			action.scoreModule = "can't verify score module name from install.yaml"
+		}
+
+		//check creator callbacks
+		if(install.files.creator != 'default') {
+			try {
+				const creator = getFileFromWebpack('creator.js').toString()
+				let missingCreatorCalls = []
+				const neededCreatorCallbacks = [
+					'initNewWidget',
+					'initExistingWidget',
+					'onMediaImportComplete',
+					'onQuestionImportComplete',
+					'onSaveClicked',
+					'onSaveComplete'
+				]
+				neededCreatorCallbacks.forEach(callback => {
+					const callbackCheck = new RegExp(`(function ${callback}){1}|(${callback} = function){1}`, 'g')
+					const callbackMatch = creator.match(callbackCheck)
+					if(!callbackMatch || callbackMatch.length > 1) {
+						status.creatorCallback = 'fail'
+						action.creatorCallback = `'${callback}' method missing or defined more than once`
+						missingCreatorCalls.push(callback)
+					}
+				})
+				const neededCreatorCoreCalls = [
+					'save',
+					// 'cancelSave',
+					'start'
+				]
+				neededCreatorCoreCalls.forEach(coreCall => {
+					const coreCallCheck = new RegExp(`Materia.CreatorCore.${coreCall}`, 'g')
+					const coreCallMatch = creator.match(coreCallCheck)
+					if(!coreCallMatch || coreCallMatch.length > 1) {
+						status.creatorCallback = 'fail'
+						action.creatorCallback = `CreatorCore '${coreCall}' method never called`
+						missingCreatorCalls.push(coreCall)
+					}
+				})
+				if(missingCreatorCalls.length == 0) status.creatorCallback = 'pass'
+			} catch(error) {
+				status.creatorCallback = 'fail'
+				action.creatorCallback = "creator source code missing or can't be parsed"
+			}
+		} else {
+			status.creatorCallback = 'pass'
+			action.creatorCallback = 'widget using default creator'
+		}
+
+		//check player callbacks
+		try {
+			const player = getFileFromWebpack('player.js').toString()
+			const playerSaveMatch = player.match(/Materia.Engine.start/g)
+			if(!playerSaveMatch || playerSaveMatch > 1) {
+				status.playerCallback = 'fail'
+				action.playerCallback = "EngineCore 'start' method missing or called more than once"
+			} else {
+				status.playerCallback = 'pass'
+			}
+		} catch(error) {
+			status.playerCallback = 'fail'
+			action.playerCallback = "player source code missing or can't be parsed"
+		}
+
+		//check score screen callbacks
+		if(install.score.score_screen) {
+			const scoreScreen = getFileFromWebpack('scorescreen.js').toString()
```

Comment:
`scorescreen.js` won't work in all situations.  In fact it's super hard to figure out what js file to load. It can be just about anything.  Perhaps we can't automate it (or maybe we need to do a little more guessing or detection).

For instance, Crossword uses `scoreScreen.js`.

Crossword's Install.yaml:
```
score:
  is_scorable: Yes
  score_module: Crossword
  score_screen: scoreScreen.html
```
which would then lead us to the js script loaded in the html...

Crossword's  scoreScreen.html:
```
<!DOCTYPE html>
<html>
	<head>
		<title>Crossword</title>
		<!-- ... -->
		<script src="scoreScreen.js" data-embed></script>
	</head>
```

That said, it may not even be in the head - and it may not be the only js file loaded.